### PR TITLE
Textfield for the quantity in Product Detail View and cartItemView + Add "added to cart" view on top of the product detail view

### LINKED ISSUE #26: Add textfield of user input for the quantity
So far, user has the option to increase and decrease the quantity of the product by pressing on the button, which later could cause inconvenience when user has a high demand of the product. We could implement the textfield so user can input their desired number.

- [x] implement textfield
- [x] add numeric pad when textfield is pressed
- [x] add a submit button to the numeric pad so numeric pad can be dismiss / lose its focus state
- [x] limit the number of products they can buy from the product detail view

### LINKED ISSUE #29: Make confirmation window to show item has been added to local shopping cart
When a user adds a product from the product detail view to local shopping cart, we need a confirmation to show the item has been successfully added to cart. A small pop up would be sufficient for this and it should automatically go away after 1 or 2 seconds. The [picture](https://github.com/algebra2boy/ShopHub/assets/103079472/908e6eef-504d-4bc2-8224-30570118a443) might help.

### LINKED ISSUE #32: Quantity number is not shown properly in cartItemView
There is a bug. A bug that is so bad that I don't know how to fix it yet, but I will take it a look when I have time. When the user updates the quantity from the product detail view the second time, the quantity in the local shopping cart is not updated correctly. The reason should be inconsistent state :(

### PR DESCRIPTION
**Pull request by algebra2boy**
Add a textfield for user desired numbers along with a numeric pad and focus state when submit button is pressed. 
- [ ] Need to fix this bug ASAP: https://github.com/algebra2boy/ShopHub/issues/32
Screenshot 2023-09-09 at 12 56 57 PM

### CODE REVIEW

**Comment by algebra2boy** [comment]
Still need to work on the blur effect on the background when "added to cart" view is shown on top of the product detail view.

File: ShopHub/ShopHub/User Interface/Page Views/Cart/CartItemView.swift [hunk 1]
```diff
@@ -54,34 +61,33 @@ struct CartItemView: View {
                     
                     HStack {
                         Button {
-                            shoppingCart.decrementQuantity(of: product)
```
Comment (TaoLyn838) [inline ShopHub/ShopHub/User Interface/Page Views/Cart/CartItemView.swift:57]: Awesome! Great optimization.

File: ShopHub/ShopHub/User Interface/Page Views/Cart/CartItemView.swift [hunk 2]
```diff
@@ -54,34 +61,33 @@ struct CartItemView: View {
...
+                        TextFieldQuantityView(value: $quantity, focusState: _isQuantityFocused)
                         
                         Button {
-                            shoppingCart.incrementQuantity(of: product)
-                            
+                            quantity += 1
                         } label: {
                             Image(systemName: "plus.square")
                         }
-                        .disabled(shoppingCart.getQuantity(of: product) == 999)
```
Comment (TaoLyn838) [inline ShopHub/ShopHub/User Interface/Page Views/Cart/CartItemView.swift:74]: GOOD!

File: ShopHub/ShopHub/User Interface/Page Views/Cart/CartItemView.swift [hunk 3]
```diff
@@ -54,34 +61,33 @@ struct CartItemView: View {
                     
                     HStack {
                         Button {
-                            shoppingCart.decrementQuantity(of: product)
+                            quantity -= 1
                         } label: {
                             Image(systemName: "minus.square")
                             
                         }
-                        .disabled(shoppingCart.getQuantity(of: product) == 1)
```
Comment (TaoLyn838) [inline ShopHub/ShopHub/User Interface/Page Views/Cart/CartItemView.swift:62]: 👍

File: ShopHub/ShopHub/User Interface/Page Views/Products/Sale/ProductDetailView.swift [hunk 4]
```diff
@@ -14,6 +14,8 @@ struct ProductDetailView: View {
     
     // Internal State
     @State private var quantity = 1
+    @State private var isAddButtonPressed = false
```
Comment (TaoLyn838) [inline ShopHub/ShopHub/User Interface/Page Views/Products/Sale/ProductDetailView.swift:17]: Good!

File: ShopHub/ShopHub/User Interface/Page Views/Cart/CartItemView.swift [hunk 5]
```diff
@@ -12,9 +12,16 @@ struct CartItemView: View {
...
     
-    // Parameters
-    let product: Product
-    let quantity: Int
+    // Internal State
+    @State private var product: Product
+    @State private var quantity: Int
+    @FocusState private var isQuantityFocused: Bool
+    
+    init(product: Product, quantity: Int) {
```
Comment (TaoLyn838) [inline ShopHub/ShopHub/User Interface/Page Views/Cart/CartItemView.swift:20]: "Smart move!!! Using the `class` character to initialize cartItem from `CartViewModel` helps set and update quantities in the shopping cart.

File: ShopHub/ShopHub/ViewModels/CartViewModel/Cartable.swift [hunk 6]
```diff
@@ -15,6 +15,13 @@ protocol Cartable {
     /// adds a specific product to the local shopping cart with quantity.
     func add(product: Product, with: Int)
     
-    // remove products from the local shopping cart.
-//    func remove
+    /// remove product from the local shopping cart.
+    func delete(product: Product)
```
Comment (TaoLyn838) [inline ShopHub/ShopHub/ViewModels/CartViewModel/Cartable.swift:19]: Good!

File: ShopHub/ShopHub/ViewModels/CartViewModel/CartViewModel.swift [hunk 7]
```diff
@@ -49,12 +49,17 @@ extension CartViewModel {
         }
     }
     
+    /// update the product quantity to local shopping cart
```
Comment (TaoLyn838) [inline ShopHub/ShopHub/ViewModels/CartViewModel/CartViewModel.swift:52]: It looks good to me.

**Review by TaoLyn838 — APPROVED** [review]

**Pull request merged**